awxkit/awxkit/cli/client.py

```diff
@@ -290,6 +290,20 @@ def parse_action(self, page, from_sphinx=False):
 
         # determine the awxkit method to call
         action = self.original_action = parsed.pop('action')
+
+        # if creation is specified with --ignore-existing, try to fetch the object instead
+        if action == 'create' and parsed.get('ignore_existing') is True:
+            _filter = UNIQUENESS_RULES.get(self.resource, ['name'])[0]
```

Comment:
I'm going to have to admit to not following.

```python
UNIQUENESS_RULES = {
    'me': ('id', 'username'),
    'users': ('id', 'username'),
    'instances': ('id', 'hostname'),
}
```

So I do `awx users create ... --ignore-existing`. The resultant `_filter` is `"id"`.

But this user doesn't exist yet, how could we possibly know what its id is?

Reply:
The answer is Ryan did it wrong and it should be `[-1]`, not `[0]` 😄

Good eye.

![image](https://user-images.githubusercontent.com/214912/66141791-4a108400-e5d2-11e9-91f3-392ebe54e2ec.png)